pkg/operators/kubeipresolver/kubeipresolver.go

```diff
@@ -134,6 +146,183 @@ func (m *KubeIPResolverInstance) EnrichEvent(ev any) error {
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "k8s", err)
+				}
+				k8sKindAcc, err := k8sSubAcc.AddSubField("kind", api.Kind_String, datasource.WithFlags(datasource.FieldFlagHidden))
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "kind", err)
+				}
+				k8sNameAcc, err := k8sSubAcc.AddSubField("name",
+					api.Kind_String,
+					datasource.WithAnnotations(map[string]string{
+						datasource.TemplateAnnotation: "pod",
+					}),
+					datasource.WithFlags(datasource.FieldFlagHidden),
+				)
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "name", err)
+				}
+				k8sNamespaceAcc, err := k8sSubAcc.AddSubField("namespace",
+					api.Kind_String,
+					datasource.WithAnnotations(map[string]string{
+						datasource.TemplateAnnotation: "namespace",
+					}),
+					datasource.WithFlags(datasource.FieldFlagHidden),
+				)
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "namespace", err)
+				}
+				k8sLabelsAcc, err := k8sSubAcc.AddSubField("labels", api.Kind_String, datasource.WithFlags(datasource.FieldFlagHidden))
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "labels", err)
+				}
+				ea := endpointAccessors{
+					root:            ep,
+					subK8sKind:      k8sKindAcc,
+					subK8sName:      k8sNameAcc,
+					subK8sNamespace: k8sNamespaceAcc,
+					subK8sLabels:    k8sLabelsAcc,
+				}
+				epAccessors[ds] = append(epAccessors[ds], ea)
+			}
+		}
+	}
+
+	inst.endpointsAccessors = epAccessors
+	return inst, nil
+}
+
+func (k *KubeIPResolver) Priority() int {
+	return Priority
+}
+
+func (m *KubeIPResolverInstance) PreStart(gadgetCtx operators.GadgetContext) error {
+	m.k8sInventory.Start()
+	return nil
+}
+
+func (m *KubeIPResolverInstance) Start(gadgetCtx operators.GadgetContext) error {
+	for ds, acc := range m.endpointsAccessors {
+		for _, a := range acc {
+			ds.Subscribe(func(source datasource.DataSource, data datasource.Data) error {
+				ips := a.root.GetSubFieldsWithTag("type:" + ipAddrType)
+				if len(ips) != 1 {
+					return fmt.Errorf("expected %d %q field, got %d", 1, ipAddrType, len(ips))
+				}
+				if ips[0].Size() != 16 {
+					return fmt.Errorf("expected %q to have 16 bytes, got %d", ipAddrType, ips[0].Size())
+				}
+
+				version := a.root.GetSubFieldsWithTag("name:version")
+				if len(version) != 1 {
+					return fmt.Errorf("expected %d %q field, got %d", 1, "version", len(version))
+				}
+
+				ip := ips[0].Get(data)
+				if ip == nil {
+					return fmt.Errorf("expected %q to have a value", ipAddrType)
+				}
+				v, err := version[0].Uint8(data)
+				if err != nil {
+					return fmt.Errorf("expected %q to have a value", "version")
+				}
+				var addrStr string
+				switch v {
+				case 4:
+					addrStr = net.IP(ips[0].Get(data)[:4]).String()
+				case 6:
+					addrStr = net.IP(ips[0].Get(data)).String()
+				default:
+					return fmt.Errorf("unsupported IP version %d", v)
+				}
```

Comment:
Given that it doesn't seem a costly task, I think it's fine to not assume the formatter already formatted all this. Doing so, we don't create dependencies between operators.

Reply:
Yeah, we could move it to `operators/common` but I am fine keeping it here as well!

Reply:
Yeah, check how it'd look like if moved into a common function 🙂

Reply:
Done.